### PR TITLE
Link against Iconv

### PR DESCRIPTION
**Pull request by klemensn**
`src/util/File.cpp` calls iconv_open(3) unconditionally on
(!WIN && !MAC && !ANDROID).

Thus all UNIX platforms will use iconv.

On OpenBSD, the build immediately fails because `<iconv.h>` is not in
the default include path.

Signed-off-by: Klemens Nanni <klemens@posteo.de>

### CODE REVIEW

**Comment by klemensn** [comment]
@metsma can you share an ETA for landing open PRs (like this one) improving/fixing stuff for distribution package maintainers and/or new libdigidocpp/DigiDoc4-Client?

Thanks in advance.

**Comment by metsma** [comment]
There is already feature freeze active. They will be approved to next release cycle

**Comment by klemensn** [comment]
I previously tested this patch on top of the 3.14.8 release which worked,
but 8c224d938d8800f222bf0a821c5f6fd9eb9a4c4e (#458) moved `File.cpp` to new`libdigidocpp_util` and thus broke this PR.

I have rebased, amended and tested it in an OpenBSD build of latest HEAD as of today (d9fd053).

**Comment by klemensn** [comment]
> There is already feature freeze active. They will be approved to next release cycle

Thanks.

Conversely, does that mean that all PRs approved as of now will land in the next libdigidocpp/DigiDoc4-Client releases?

**Comment by metsma** [comment]
Yes

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -165,12 +165,20 @@ target_compile_definitions(digidocpp_priv PUBLIC
     $<$<CXX_COMPILER_ID:Clang>:XSD_CXX11_TEMPLATE_ALIAS>
 )
 
+target_include_directories(digidocpp_util PUBLIC
+    ${Iconv_INCLUDE_DIR}
+)
+
```
Comment (metsma) [inline src/CMakeLists.txt:171]: Is this actually needed?
target_link_libraries(digidocpp_util Iconv::Iconv)
will include also include directories

**Review by metsma — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (klemensn) [reply src/CMakeLists.txt]: Ah yes, also works without.

I think its a left-over from my first diff which then stuck across the rebase from v3.14.8 onto latest HEAD.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -231,6 +239,9 @@ set_target_properties(digidocpp PROPERTIES
 
 target_include_directories(digidocpp PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries(digidocpp PRIVATE ${CMAKE_DL_LIBS} minizip digidocpp_priv)
+if(NOT WIN32 AND NOT APPLE AND NOT ANDROID)
+    target_link_libraries(digidocpp PUBLIC Iconv::Iconv)
```
[the remark this answers is not in the document]
Comment (klemensn) [reply src/CMakeLists.txt]: Of course.

Builds and works as expected on OpenBSD.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -171,6 +171,10 @@ target_include_directories(digidocpp_priv PUBLIC
     ${XSD_INCLUDE_DIR}
 )
 
+target_link_libraries(digidocpp_util
```
Comment (metsma) [inline src/CMakeLists.txt:174]: I think you are missing some conditions here, thats why IOS builds are failing

**Comment by metsma** [comment]
Is still needed with 3692eaccdd4e79aa1133ba3addc6d26bebbe8672 change

**Comment by klemensn** [comment]
> Is still needed with [3692eac](https://github.com/open-eid/libdigidocpp/commit/3692eaccdd4e79aa1133ba3addc6d26bebbe8672) change

I'll rebase it and look into CI failures soon, hoping to get this merged soon so I can stop applying it locally in the OpenBSD port.

File: src/CMakeLists.txt (hunk 3 above)
Comment (klemensn) [reply src/CMakeLists.txt]: I've added the condition used in the other hunk, let's see what CI says.

**Comment by metsma** [comment]
I think this is not needed anymore. Least linux does not require it

**Comment by klemensn** [comment]
Oh I see, 3692eaccdd4e79aa1133ba3addc6d26bebbe8672 completely removed the iconv code, but that commit is only in master and not the 3.14.12 release.

**Pull request closed without merging**